### PR TITLE
[CTRL-2] Control MXControl in Hipchat

### PR DESCRIPTION
**Pull request by fcamilleri22**
Should do stuff with things.

### CODE REVIEW

File: NOTETOFRANK [hunk 1]
```diff
@@ -0,0 +1,2 @@
+run with  node-lambda run -j event_up_inst_BE_demo.json
```
Comment (fcamilleri22) [inline NOTETOFRANK]: ill delete this

File: context.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (fcamilleri22) [inline context.json]: and this

File: event_up_inst_BE_demo.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{
```
Comment (fcamilleri22) [inline event_up_inst_BE_demo.json]: where should an edited version of this example from hipchat be kept?

File: sample.json [hunk 4]
```diff
@@ -0,0 +1 @@
+"event": "room_message", "item": {"message": {"date": "2017-10-05T20:40:47.901821+00:00", "from": {"id": 2256873, "links": {"self": "https://api.hipchat.com/v2/user/2256873"}, "mention_name": "FrankCamilleri", "name": "Frank Camilleri", "version": "2WV1JEMZ"}, "id": "3b79379f-96b0-4f9e-b894-1b1536549d02", "mentions": [], "message": "/up inst MXBackend-demo", "type": "message"}, "room": {"id": 4216033, "is_archived": false, "links": {"members": "https://api.hipchat.com/v2/room/4216033/member", "participants": "https://api.hipchat.com/v2/room/4216033/participant", "self": "https://api.hipchat.com/v2/room/4216033", "webhooks": "https://api.hipchat.com/v2/room/4216033/webhook"}, "name": "MXControl", "privacy": "private", "version": "ZSL79NV5"}}, "oauth_client_id": "5912e25d-6dee-46fc-b1ab-34784b02faeb", "webhook_id": 19298437}
```
Comment (fcamilleri22) [inline sample.json]: will delete this too.

File: events/mxcontrol.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        let controlTask = {
+            action: action
+        };
+
+        //Adapted from MXControl/CLI.js
+        if (size) controlTask = Object.assign({size: size}, controlTask);
+
+        this.task =
+        (targetType == "env" || targetType == "environment")
+            ? this.task = Object.assign({environment: target}, controlTask)
```
Comment (edobry) [inline events/mxcontrol.js]: don't need to duplicate any part of this line besides the target object

File: events/mxcontrol.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+        let controlTask = {
+            action: action
+        };
+
+        //Adapted from MXControl/CLI.js
+        if (size) controlTask = Object.assign({size: size}, controlTask);
+
+        this.task =
+        (targetType == "env" || targetType == "environment")
```
Comment (edobry) [inline events/mxcontrol.js]: can do `["env", "environment"].includes(targetType)` instead of the double test

File: handlers/mxcontrol.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+const handleMXControlTask = (event) => {
+    const task = event.task;
+    const targetName = task.instance || task.environment || task.database;
+
+    const logline =
+        MXControl.buildLog(targetName, task.action, task.size, task.database);
+
+    msgMXControlRoom(logline);
+
+    return Promise.resolve(MXControl.runTask(task).then((res)=>{
```
Comment (edobry) [inline handlers/mxcontrol.js]: pls format better:
   - no parens around single param
   - spaces around `=>`
   - no empty line on function start

File: handlers/mxcontrol.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+        const action = task.action.toLowerCase();
+
+        if (action == "status" || action == "info") {
```
Comment (edobry) [inline handlers/mxcontrol.js]: `["status", "info"].includes(action)`

File: handlers/mxcontrol.js [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+            console.log(res);
+            msgMXControlRoom(
+                JSON.stringify(res)
```
Comment (edobry) [inline handlers/mxcontrol.js]: pls template string markdown pls

File: handlers/mxcontrol.js [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+    }));
+
+};
+
+module.exports = {
+    matches: event => true,
+    name: "MXControl",
+    accepts: MXControlEvent,
+    handle: handleMXControlTask,
+    irrelevantMessage: "see what happens when Frank farts..."
```
Comment (edobry) [inline handlers/mxcontrol.js]: um

File: hooks/hipchat.js [hunk 11]
```diff
@@ -0,0 +1,48 @@
+
```
Comment (edobry) [inline hooks/hipchat.js]: no empty line pls

File: hooks/hipchat.js [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+    const events = [];
+
+    if (simpleTrigger.event == "room_message") {
+        const commandArgs = simpleTrigger.message.split(" ");
+
+        if (commandArgs[0] == "/mxcontrol")
+            events.push(new MXControlEvent(
+                trigger,
+                commandArgs[1], //action
```
Comment (edobry) [inline hooks/hipchat.js]: maybe turn these into an object, so its clear what each param means without needing comments?

File: hooks/hipchat.js [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+                commandArgs[2], //targetType
+                commandArgs[3], //target
+                commandArgs[4]  //size
+            ));
+    }
+
+    return events;
+};
+
+const simplifyHipchatTrigger = (trigger) => {
```
Comment (edobry) [inline hooks/hipchat.js]: can destructure instead of having this function

File: package.json [hunk 14]
```diff
@@ -13,6 +13,7 @@
     "markdown2confluence-cws": "^2.0.2",
     "marked": "^0.3.6",
     "moment": "^2.18.1",
+    "mxcontrol": "github:massexchange/mxcontrol",
```
Comment (edobry) [inline package.json]: npm package pls

**Review by edobry — CHANGES_REQUESTED** [review]
mostly good, few small things.

File: hooks/hipchat.js (hunk 12 above)
Comment (fcamilleri22) [reply hooks/hipchat.js]: Will do when I come back

File: hooks/hipchat.js (hunk 13 above)
Comment (fcamilleri22) [reply hooks/hipchat.js]: Will do when I come back

File: handlers/mxcontrol.js (hunk 10 above)
Comment (fcamilleri22) [reply handlers/mxcontrol.js]: WELL DO YOU WANNA?

File: handlers/mxcontrol.js (hunk 9 above)
Comment (fcamilleri22) [reply handlers/mxcontrol.js]: Will do when I come back

File: handlers/mxcontrol.js (hunk 7 above)
Comment (fcamilleri22) [reply handlers/mxcontrol.js]: The robut that was supposed to take care of it suffered from an insane attack of existential woe when I told it that that was its sole purpose for existing.

(i'll take care of it more completely when i get back

File: package.json (hunk 14 above)
Comment (fcamilleri22) [reply package.json]: Will do when I come back

File: src/handlers/mxcontrol.js [hunk 15]
```diff
@@ -0,0 +1,69 @@
+const nconf = require("nconf");
```
Comment (fcamilleri22) [inline src/handlers/mxcontrol.js:1]: Will be rewriting this file when I get back

File: src/events/mxcontrol.js [hunk 16]
```diff
@@ -0,0 +1,31 @@
+const Event = require("../event");
```
Comment (fcamilleri22) [inline src/events/mxcontrol.js:1]: Will be rewriting this file when I get back

File: src/hooks/hipchat.js [hunk 17]
```diff
@@ -0,0 +1,50 @@
+const Hook = require("../hook");
```
Comment (fcamilleri22) [inline src/hooks/hipchat.js:1]: Stable. Will only be editing line 38 in service of redoing mxcontrol related files.

**Comment by fcamilleri22** [comment]
Closing PR because this is outdated. See new PR

**Pull request closed without merging**